app/src/checkupdatesdialog.cpp

```diff
@@ -0,0 +1,193 @@
+#include "checkupdatesdialog.h"
+#include <QNetworkReply>
+#include <QNetworkAccessManager>
+#include <QHBoxLayout>
+#include <QVBoxLayout>
+#include <QLabel>
+#include <QProgressBar>
+#include <QPushButton>
+#include <QDesktopServices>
+
+#include "util.h"
+
+
+CheckUpdatesDialog::CheckUpdatesDialog()
+{
+    Qt::WindowFlags eFlags = Qt::Dialog | Qt::WindowTitleHint | Qt::WindowCloseButtonHint;
+    setWindowFlags(eFlags);
+    setMinimumSize(QSize(400, 150));
+
+    QLabel* logoLabel = new QLabel;
+    logoLabel->setPixmap(QPixmap(":/icons/logo.png"));
+    logoLabel->setFixedSize(QSize(72, 72));
+
+    mTitleLabel = new QLabel;
+    mTitleLabel->setText(tr("Checking for Updates...", "status description in the check-for-update dialog"));
+
+    mDetailLabel = new QLabel;
+    mDetailLabel->setWordWrap(true);
+
+    //If minimum and maximum both are set to 0, the bar shows a busy indicator instead of a percentage of steps. 
+    mProgressBar = new QProgressBar;
+    mProgressBar->setMaximum(0);
+    mProgressBar->setMinimum(0);
+    mProgressBar->setValue(0);
+
+    mDownloadButton = new QPushButton(tr("Download"));
+    mCloseButton = new QPushButton(tr("Close"));
+
+    QHBoxLayout* hButtonLayout = new QHBoxLayout;
+    hButtonLayout->addWidget(mDownloadButton);
+    hButtonLayout->addWidget(mCloseButton);
+    
+    QVBoxLayout* vLayout = new QVBoxLayout;
+    vLayout->addWidget(mTitleLabel);
+    vLayout->addWidget(mDetailLabel);
+    vLayout->addWidget(mProgressBar);
+    vLayout->addSpacerItem(new QSpacerItem(20, 20, QSizePolicy::Minimum, QSizePolicy::Expanding));
+    vLayout->addLayout(hButtonLayout);
+
+    QHBoxLayout* mainLayout = new QHBoxLayout;
+    mainLayout->addWidget(logoLabel);
+    mainLayout->addLayout(vLayout);
+    setLayout(mainLayout);
+
+    mDownloadButton->setDisabled(true);
+
+    connect(mDownloadButton, &QPushButton::clicked, this, &CheckUpdatesDialog::gotoDownloadPage);
+    connect(mCloseButton, &QPushButton::clicked, this, &CheckUpdatesDialog::closeDialog);
+}
+
+CheckUpdatesDialog::~CheckUpdatesDialog()
+{
+    mNetworkManager->deleteLater();
+}
+
+void CheckUpdatesDialog::startChecking()
+{   
+#ifdef NIGHTLY
+    nightlyBuildCheck();
```

Comment:
yeah, I am thinking how to do it. It means we have to give nightly build a proper version number which can compare to a regular build.

Reply:
We could use the Google Drive API to get the files currently in the drive folder. The request would look like this:
```
https://www.googleapis.com/drive/v3/files?q='<FOLDER_ID>'+in+parents&key=<API_KEY>
```
The result is a json file that looks like this:
```json
{
 "kind": "drive#fileList",
 "incompleteSearch": false,
 "files": [
  {
   "kind": "drive#file",
   "id": "1IZgGuJmy4JOnAdEQEkjHCv8qVGFqvcpr",
   "name": "pencil2d-mac-2018-11-13.zip",
   "mimeType": "application/zip"
  },
  {
   "kind": "drive#file",
   "id": "1HQhIrf97nTIfpsN9i4NqhROOlCZKWld7",
   "name": "pencil2d-mac-2018-11-04.zip",
   "mimeType": "application/zip"
  },
  ...
 ]
}
```
The v2 api provides even more information if you need it, but we should just be able to parse the name and compare it to the date stored in the build.

A side note, I discovered during researching this that Google Drive does have a "revisions" feature which automatically deletes old revisions after a month or 100 revisions and keeps the same link so you could have a "latest nightly build" link for each OS which is pretty nice. Unfortunately it doesn't seem that you can access the revision list without OAuth, so if we're going to add a nightly build check with the API, it would not be a good idea to switch to this.